Tasks/AzureMysqlDeployment/models/AzureMysqlTaskParameter.ts

```diff
@@ -0,0 +1,97 @@
+import tl = require("vsts-task-lib/task");
+
+export class AzureMysqlTaskParameter {
+	
+	private connectedServiceName: string; 
+	private serverName: string;
+	private databaseName: string;
+	private sqlUserName: string;
+	private sqlPassword: string;
+	private taskNameSelector: string;
+	private sqlFile: string;
+	private sqlInline: string;
+	private sqlAdditionalArguments: string;
+	private inlineAdditionalArguments;
+	private ipDetectionMethod: string;
+	private startIpAddress: string;
+	private endIpAddress: string;
+	private deleteFirewallRule: boolean;
+
+    constructor() {
+        try {
+            this.connectedServiceName = tl.getInput('ConnectedServiceName', true);
+            this.serverName = tl.getInput('ServerName', true);
+            this.databaseName = tl.getInput('DatabaseName', false);
+            this.sqlUserName = tl.getInput('SqlUsername', true);
+            this.sqlPassword = tl.getInput('SqlPassword', true);
+            this.taskNameSelector = tl.getInput('TaskNameSelector', true);
+            this.sqlFile = tl.getInput('SqlFile', false);
+            this.sqlInline = tl.getInput('SqlInline', false);
+            this.sqlAdditionalArguments = tl.getInput('SqlAdditionalArguments', false);
+            this.inlineAdditionalArguments = tl.getInput('InlineAdditionalArguments', false);
+            this.ipDetectionMethod  = tl.getInput('IpDetectionMethod', false);
+            this.startIpAddress = tl.getInput('StartIpAddress', false);
+            this.endIpAddress = tl.getInput('EndIpAddress', false);
+            this.deleteFirewallRule = tl.getBoolInput('DeleteFirewallRule', false);
+        }
+        catch (error) {
+            throw new Error(tl.loc("ARGD_ConstructorFailed", error.message));
```

Comment:
It's not good practice to throw an exception from a constructor. Is same approach followed in other tasks.

Reply:
It is same approach we are following at other place . If we will not able to initialized object then we should throw error from constructor.